packages/zwave-js/src/lib/commandclass/HumidityControlSetpointCC.ts

```diff
@@ -0,0 +1,939 @@
+			case "setpoint":
+				if (typeof propertyKey !== "number") {
+					throw new ZWaveError(
+						`${
+							CommandClasses[this.ccId]
+						}: "${property}" must be further specified by a numeric property key`,
+						ZWaveErrorCodes.Argument_Invalid,
+					);
+				}
+
+				return (await this.get(propertyKey))?.value;
+			default:
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+	public async get(setpointType: HumidityControlSetpointType) {
+		this.assertSupportsCommand(
+			HumidityControlSetpointCommand,
+			HumidityControlSetpointCommand.Get,
+		);
+
+		const cc = new HumidityControlSetpointCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			setpointType,
+		});
+		const response =
+			await this.driver.sendCommand<HumidityControlSetpointCCReport>(
+				cc,
+				this.commandOptions,
+			);
+		if (!response) return;
+		return response.type === HumidityControlSetpointType["N/A"]
+			? // not supported
+			  undefined
+			: // supported
+			  {
+					value: response.value,
+					scale: response.scale,
+			  };
+	}
+
+	public async set(
+		setpointType: HumidityControlSetpointType,
+		value: number,
+		scale: number,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			HumidityControlSetpointCommand,
+			HumidityControlSetpointCommand.Set,
+		);
+
+		const cc = new HumidityControlSetpointCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			setpointType,
+			value,
+			scale,
+		});
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+	public async getCapabilities(setpointType: HumidityControlSetpointType) {
+		this.assertSupportsCommand(
+			HumidityControlSetpointCommand,
+			HumidityControlSetpointCommand.CapabilitiesGet,
+		);
+
+		const cc = new HumidityControlSetpointCCCapabilitiesGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			setpointType,
+		});
+		const response =
+			await this.driver.sendCommand<HumidityControlSetpointCCCapabilitiesReport>(
+				cc,
+				this.commandOptions,
+			);
+		if (response) {
+			return pick(response, [
+				"minValue",
+				"maxValue",
+				"minValueScale",
+				"maxValueScale",
+			]);
+		}
+	}
+
+	public async getSupportedSetpointTypes(): Promise<
+		readonly HumidityControlSetpointType[] | undefined
+	> {
+		this.assertSupportsCommand(
+			HumidityControlSetpointCommand,
+			HumidityControlSetpointCommand.SupportedGet,
+		);
+
+		const cc = new HumidityControlSetpointCCSupportedGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response =
+			await this.driver.sendCommand<HumidityControlSetpointCCSupportedReport>(
+				cc,
+				this.commandOptions,
+			);
+		return response?.supportedSetpointTypes;
+	}
+
+	public async getScaleSupported(
```

Comment:
```suggestion
	public async getSupportedScales(
```
I prefer using more "natural" method names for the CC API.